tests/test_sampler.py

```diff
@@ -116,13 +123,72 @@ def test_rate_limiting_sampler():
         for i in range(0, 3):
             sampled, tags = sampler.is_sampled(0)
             assert not sampled, 'but no further, since time is stopped'
-        assert tags == {
-            'sampler.type': 'ratelimiting',
-            'sampler.param': 2,
-        }
+        assert tags == get_tags('ratelimiting', 2)
     sampler.close()
     assert '%s' % sampler == 'RateLimitingSampler(2)'
 
+    # Test with rate limit of less than 1 second
+    sampler = RateLimitingSampler(0.1)
+    ts = time.time()
+    sampler.last_tick = ts
+    with mock.patch('jaeger_client.sampler.RateLimitingSampler.timestamp') \
+            as mock_time:
+        mock_time.side_effect = lambda: ts  # always return same time
+        assert sampler.timestamp() == ts
+        sampled, _ = sampler.is_sampled(0)
+        assert not sampled
+
+        # move time 110ms forward, enough credits to pay for one sample
+        mock_time.side_effect = lambda: ts + 0.11
+        assert sampler.is_sampled(0)
+    sampler.close()
+    assert '%s' % sampler == 'RateLimitingSampler(0.1)'
+
+
+def test_guaranteed_throughput_probabilistic_sampler():
+    sampler = GuaranteedThroughputProbabilisticSampler('op', 2, 0.5)
+    id1 = 1L << 63  # second most significant bit, dividing full range in half
+    assert id1 == 0x8000000000000000L
+    sampled, tags = sampler.is_sampled(id1-10)
+    assert sampled
+    assert tags == get_tags('probabilistic', 0.5)
+    sampled, tags = sampler.is_sampled(id1+10)
+    assert sampled
+    assert tags == get_tags('lowerbound', 0.5)
+    sampled, _ = sampler.is_sampled(id1+10)
+    assert not sampled
+
+    sampler.close()
+    assert '%s' % sampler == 'GuaranteedThroughputProbabilisticSampler(op, 0.5, 2)'
+
+
+def test_adaptive_sampler():
+    sampling_rates = [
+        OperationSamplingStrategy('op', ProbabilisticSamplingStrategy(0.5))
+    ]
+    strategies = PerOperationSamplingStrategies(0.51, 3, sampling_rates)
+
+    sampler = AdaptiveSampler(strategies, 2)
+    id1 = 1L << 63  # second most significant bit, dividing full range in half
+    assert id1 == 0x8000000000000000L
```

Comment:
we keep doing this, let's define  constant once and have a standalone test that validates that the constant is equal to `1L << 63` expression